### ISSUE TITLE
Blog post: Pros, cons and best practices of using Jupyter Notebooks

### ISSUE THREAD

**Issue by KirstieJane** [the-turing-way/the-turing-way#68]
This issue is to capture the good points and the bad points of jupyter notebooks.

* debugging and version control is hard
* could you be doing this elsewhere?
* need to run in a set order (not always done!)

Talk at jupyter con: https://conferences.oreilly.com/jupyter/jup-ny/public/schedule/detail/68282

Nice contrast to perl (spelling?) in Rmarkdown that converts Rmarkdown files to scripts for more generalised development.

Important to note that binder doesn't *require* you to use jupyter notebooks - task for @sgibson91 to highlight how to use binder with other (better?) ways of working.

**Comment by sgibson91** [comment]
@iamleeg and I have been discussing pros/cons/best practices of using jupyter notebooks in data science/research and would like to write a blog post. Then I had the brilliant idea of putting this blog post in The Turing Way! So if it's ok, we might hijack this issue to plan that post 😄

**Comment by sgibson91** [comment]
@iamleeg I think we should go down the above proposed route of "when is it appropriate to use a Jupyter Notebook for research", note what notebooks do well and what features might catch people out, and then compile a list of useful extensions that solve (or attempt to solve) those gotchas. What do you think?

**Comment by iamleeg** [comment]
@sgibson91 Yes, that thing! "Here are some justified good practices when using Jupyter for research" is better than "yah notebooks suck urrr"
